### ISSUE TITLE
Indentation: false positives when switch expression is invocation target

### ISSUE THREAD

**Issue by ksobolew** [checkstyle/checkstyle#11741]
I have read check documentation: https://checkstyle.org/config_xxxxxx.html#NameOfAffectedCheck
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
%  javac scratch_1.java

%  cat config.xml 
<?xml version="1.0" ?>
<!DOCTYPE module PUBLIC "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN" "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
    <!-- Annotations -->
    <!-- Allow suppressing with an annotation -->
    <module name="TreeWalker">
        <module name="Indentation"/>
    </module>
</module>

%  cat scratch_1.java 
class Scratch
{
    public static void main(String[] args)
    {
        (switch (args.length) {
            case 1 -> args[0];
            case 2 -> args[1];
            default -> throw new IllegalArgumentException();
        }).length();

        String s = switch (args.length) {
            case 1 -> args[0];
            case 2 -> args[1];
            default -> throw new IllegalArgumentException();
        };
        s.length();
    }
}

%  RUN_LOCALE="-Duser.language=en -Duser.country=US"

%  java -jar checkstyle-10.3-all.jar -c config.xml scratch_1.java
Starting audit...
[ERROR] /Users/kudivuhadi/Library/Application Support/JetBrains/IntelliJIdea2022.2/scratches/scratch_1.java:6:13: 'case' child has incorrect indentation level 12, expected level should be 16. [Indentation]
[ERROR] /Users/kudivuhadi/Library/Application Support/JetBrains/IntelliJIdea2022.2/scratches/scratch_1.java:7:13: 'case' child has incorrect indentation level 12, expected level should be 16. [Indentation]
[ERROR] /Users/kudivuhadi/Library/Application Support/JetBrains/IntelliJIdea2022.2/scratches/scratch_1.java:8:13: 'case' child has incorrect indentation level 12, expected level should be 16. [Indentation]
[ERROR] /Users/kudivuhadi/Library/Application Support/JetBrains/IntelliJIdea2022.2/scratches/scratch_1.java:9:9: 'switch rcurly' has incorrect indentation level 8, expected level should be 12. [Indentation]
Audit done.
Checkstyle ends with 4 errors.
```

---

The `main` method contains two switch expressions: the first is immediately used as a target of method invocation, the second does the same, but its value is first assigned to an intermediate variable. `Indentation` reports errors for the first switch, but not the second. This is similar to #8929, which (I believe) fixed it for the second switch.

**Comment by ksobolew** [comment]
I also just noticed that the same thing happens when the switch is `yield`ded from an outer switch:

```java
int result = switch (something) {
    case CASE -> {
        yield switch (somethingElse) {
            case CASE -> 1;
            default -> 2;
        }
    };
    default -> 3;
}
```

**Comment by DmitriySerobyan** [comment]
I have the same problem

**Comment by JarvisCraft** [comment]
Having the same issue when using Checkstyle with Java 17 sources containing new switch expression constructs.

**Comment by SharplEr** [comment]
On my work project we use JDK 17, and also faced with that problem with nested switch expressions.

**Comment by sherman** [comment]
Voted for this issue. Switched to JDK 17 where the switch expression significantly improved and used widely.

**Comment by Kevin222004** [comment]
I am on it.

**Comment by ZaheerAhmadDev** [comment]
I am on it.
